### PR TITLE
bugfix for type of some output files in cnv_somatic_pair_workflow.wdl

### PR DESCRIPTION
**Pull request by slzhao**
Hello, 

I made two changes in output files of cnv_somatic_pair_workflow.wdl to fix bugs:
1. Change "File" into "String" for all "entity_id" outputs. The "File" type will make error when Output Copying was used in cromwell, as there is no such file in output;
2. Change "File select_first([CNVOncotatorWorkflow.oncotated_called_file, "null"])" into "File? oncotated_called_file_tumor" for oncotate and funcotate outputs. If oncotate or funcotate was not performed, the original will output "null" and make error when Output Copying was used in cromwell, as there is "null" file in output;

### CODE REVIEW

**Comment by cmnbroad** [comment]
I've encountered and reported a similar problem for cromwell. I think the changes proposed here won't result in correct localization/delocalization, but I'm not sure what the correct solution is for optional outputs where no file name is provided. See [this ticket](https://broadworkbench.atlassian.net/projects/BA/issues/BA-6530?filter=allopenissues&orderby=priority%20DESC&keyword=optional%20outputs) in the cromwell jira.

**Comment by lbergelson** [comment]
I'm not sure external people can see that jira at all.

**Comment by slzhao** [comment]
Hello,

Just want to clarify that the changes I made is about "File?" in output and it works with cromwell v51. If the task generated that file was not run (such as skipped by `if` in this cnv_somatic_pair_workflow.wdl), the output will also skip it and will not report error.

I think the ticket you mentioned is talking about "String?", which is different that "File?" in output?

Thanks!

**Comment by droazen** [comment]
@mwalker174 Can you review this one when you get a chance?

File: scripts/cnv_wdl/somatic/cnv_somatic_pair_workflow.wdl [hunk 1]
```diff
@@ -568,10 +568,15 @@ workflow CNVSomaticPairWorkflow {
...
 
-        File oncotated_called_file_tumor = select_first([CNVOncotatorWorkflow.oncotated_called_file, "null"])
-        File oncotated_called_gene_list_file_tumor = select_first([CNVOncotatorWorkflow.oncotated_called_gene_list_file, "null"])
-        File funcotated_called_file_tumor = select_first([CNVFuncotateSegmentsWorkflow.funcotated_seg_simple_tsv, "null"])
-        File funcotated_called_gene_list_file_tumor = select_first([CNVFuncotateSegmentsWorkflow.funcotated_gene_list_tsv, "null"])
+        # File? oncotated_called_file_tumor = select_first([CNVOncotatorWorkflow.oncotated_called_file, "null"])
+        # File? oncotated_called_gene_list_file_tumor = select_first([CNVOncotatorWorkflow.oncotated_called_gene_list_file, "null"])
+        # File? funcotated_called_file_tumor = select_first([CNVFuncotateSegmentsWorkflow.funcotated_seg_simple_tsv, "null"])
+        # File? funcotated_called_gene_list_file_tumor = select_first([CNVFuncotateSegmentsWorkflow.funcotated_gene_list_tsv, "null"])
+
```
Comment (mwalker174) [inline scripts/cnv_wdl/somatic/cnv_somatic_pair_workflow.wdl:575]: You can delete these commented lines

**Review by mwalker174 — APPROVED** [review]
Thank you @slzhao this looks good. Please see my one comment below.

**Comment by samuelklee** [comment]
Thanks for catching these, @slzhao!

**Comment by ldgauthier** [comment]
@slzhao do you have time to make that one small edit?  After that, this is good to merge.

**Comment by samuelklee** [comment]
@slzhao item 1 in your original post randomly came up in the OpenWDL Slack, which ultimately caused me to realize that this PR is still open. Would you like to remove the commented lines and merge? If we don't hear back from you, I'll merge this and issue a separate PR to remove the commented lines.

**Pull request merged**